utils/tools.go

```diff
@@ -76,13 +76,28 @@ func AddressTo32Bytes(address common.Address) [32]byte {
 	addressBytes := address.Bytes()
 	address32Byte := [32]byte{}
 	for i := 1; i <= common.AddressLength; i++ {
-
 		address32Byte[32-i] = addressBytes[common.AddressLength-i]
 	}
 	return address32Byte
 
 }
 
+// ByteArrayTo32BytesLeftPadded converts an address to 32 a byte array
+// The length of the input has to be less or equals to 32
+func ByteArrayTo32BytesLeftPadded(in []byte) ([32]byte, error) {
+	byte32 := [32]byte{}
+	if len(in) > 32 {
+		return byte32, errors.New("incorrect input length %d should be 32", len(in))
+	}
+	if len(in) == 0 {
+		return byte32, nil
+	}
+	for i := 1; i <= len(in); i++ {
+		byte32[32-i] = in[len(in)-i]
```

Comment:
I feel this can be simplified.

pseudo code
```
toBepadded := 32 - len(in)
out := append(make([]byte, toBePadded, toBePadded), in...)
return utils.SliceToByte32(out)
```